cli/prisma2/src/bin.ts

```diff
@@ -120,21 +120,38 @@ if (require.main === module) {
         process.exit(code)
       }
     })
-    .catch(err => {
-      if (err.rustStack) {
-        handlePanic(err, packageJson.name, packageJson.version).catch(e => {
+    .catch((err) => {
+      let shouldExit = false
+
+      function handleIndividualError(err) {
+        if (err.rustStack) {
+          handlePanic(err, packageJson.name, packageJson.version).catch(e => {
+            if (debugLib.enabled('prisma')) {
+              console.error(chalk.redBright.bold('Error: ') + e.stack)
+            } else {
+              console.error(chalk.redBright.bold('Error: ') + e.message)
+            }
+          })
+        } else {
+          shouldExit = true
           if (debugLib.enabled('prisma')) {
-            console.error(chalk.redBright.bold('Error: ') + e.stack)
+            console.error(chalk.redBright.bold('Error: ') + err.stack)
           } else {
-            console.error(chalk.redBright.bold('Error: ') + e.message)
+            console.error(chalk.redBright.bold('Error: ') + err.message)
           }
-        })
-      } else {
-        if (debugLib.enabled('prisma')) {
-          console.error(chalk.redBright.bold('Error: ') + err.stack)
-        } else {
-          console.error(chalk.redBright.bold('Error: ') + err.message)
         }
+      }
+
+      // Sindre's pkg p-map & co are using AggregateError, it is an iterator.
+      if (typeof err[Symbol.iterator] === 'function') {
+        for (const individualError of err) {
+          handleIndividualError(individualError)
+        }
+      } else {
+        handleIndividualError(err)
+      }
+
+      if (shouldExit) {
```

Comment:
Why do we need `shouldExit`? Shouldn't it always exit?

Reply:
So this logic was here before, my understanding is that when we have a err.rustPanic we delegate this to this custom interactive logic?